packages/rulesets/src/spectral/functions/validate-patch-body-param-properties.ts

```diff
@@ -46,19 +46,39 @@ export const validatePatchBodyParamProperties = createRulesetFunction<unknown, O
         _opts.should.forEach((p: string) => {
           if (!getProperties(bodyParameter)?.[p] && getProperties(responseSchema)?.[p]) {
             errors.push({
-              message: `The patch operation body parameter schema should contains property '${p}'.`,
+              message: `The patch operation body parameter schema should contain property '${p}'.`,
               path: [...path, "parameters", index],
             })
           }
         })
       }
       if (_opts.shouldNot) {
         _opts.shouldNot.forEach((p: string) => {
-          if (getProperties(bodyParameter)?.[p]) {
-            errors.push({
-              message: `The patch operation body parameter schema should not contains property ${p}.`,
-              path: [...path, "parameters", index],
-            })
+          const property = getProperties(bodyParameter)?.[p];
+          if (property) {
+            let isPropertyBoolean = false;
+            let isPropertyImmutable = false;
+
+            if (property["readOnly"] && property["readOnly"] === true)
+            {
+              isPropertyBoolean = true;
+            }
+
+            if (property["x-ms-mutability"] && Array.isArray(property["x-ms-mutability"])) {
+                const schemeArray = property["x-ms-mutability"];
+                if (!schemeArray.includes("update"))
+                {
+                    isPropertyImmutable = true;
+                }
+            }
+
+            if (!isPropertyBoolean && !isPropertyImmutable)
```

Comment:
```suggestion
            if (!(isPropertyReadOnly || isPropertyImmutable))
```

I think this is easier to reason about.

Reply:
I felt the other way looks easier :)

Reply:
fixed anyway

Reply:
I have one more idea here, as explained [in this comment](https://github.com/Azure/azure-openapi-validator/pull/513#discussion_r1196090530).